### ISSUE TITLE
Interruption problem in IOS devices

### ISSUE THREAD

**Issue by AliShehab93** [ryanheise/audio_session#62]
Hi Ryanheise,
I am facing a problem in an audio session listening for interruptions in ios. In android, streaming is working properly, while in ios it has a delay of 1sec 2sec to reach the pause case. And when resume the record after an interruption is ended, it will replace the paused recording while this issue is working perfectly in android. Also, sometimes, before pause interruption in ios, let us say I record for about 8 sec then a call received, here record is paused and when finishing interruption then listen to the saved record, it will be for about 3 sec recorded not even 7 sec since always it makes a delay as 1 sec 2 sec. 
Have you figured out these issues before?

**Comment by AliShehab93** [comment]
Yes I tried the phone call interruption, I didn’t try other interruptions since I need others right now.

**Comment by ryanheise** [comment]
I don't see anything in the code that could cause a delay, so maybe this is just the normal behaviour of iOS (you could confirm by editing the native code and inserting a log when the interruption is received natively).

**Comment by AliShehab93** [comment]
I don't think it is a normal behavior of ios since it is working with other apps. I may create a new branch from your package and debug this scenario in the native part to the flutter part and I will discuss with you if the problem is the same.

**Comment by chashkovdaniil** [comment]
> I don't think it is a normal behavior of ios since it is working with other apps. I may create a new branch from your package and debug this scenario in the native part to the flutter part and I will discuss with you if the problem is the same.

Hi! Have you been able to solve this problem?

**Comment by AliShehab93** [comment]
Hello dear,

Actually I didn’t test it yet, and I don’t think so it os solved. Are you
facing the same issue as mine?

**Comment by chashkovdaniil** [comment]
> 

Yes, but in flutter_sound package

**Comment by ryanheise** [comment]
I'd be interested to know what the logging reveals.

**Comment by vikasdhaka16** [comment]
> Hi Ryanheise, I am facing a problem in an audio session listening for interruptions in ios. In android, streaming is working properly, while in ios it has a delay of 1sec 2sec to reach the pause case. And when resume the record after an interruption is ended, it will replace the paused recording while this issue is working perfectly in android. Also, sometimes, before pause interruption in ios, let us say I record for about 8 sec then a call received, here record is paused and when finishing interruption then listen to the saved record, it will be for about 3 sec recorded not even 7 sec since always it makes a delay as 1 sec 2 sec. Have you figured out these issues before?

Hey in my ios also interuption is not working but in android it’s working
